### PR TITLE
Add iml-postgres crate

### PR DESCRIPTION
**Pull request by jgrund**
Add a new crate so services can utilize postgres
within rust.

Signed-off-by: Joe Grund <jgrund@whamcloud.io>

### CODE REVIEW

File: iml-agent/Cargo.toml [hunk 1]
```diff
@@ -21,12 +21,12 @@ spinners = "1.0.0"
 libc = "0.2.58"
 log = "0.4.6"
 prettytable-rs = "0.8"
-reqwest = { version = "0.9.17", features = ["default-tls", "tls"] }
+reqwest = { version = "0.9.18", features = ["default-tls", "tls"] }
```
Comment (utopiabound) [inline iml-agent/Cargo.toml]: This conflicts with #998.  Other patches sets version to "0.9"

File: iml-agent/Cargo.toml [hunk 2]
```diff
@@ -21,12 +21,12 @@ spinners = "1.0.0"
...
 hyper = "0.12"
 http = "0.1"
 serde = { version = "1", features = ["derive"] }
 serde_json = "1"
-structopt = "0.2.17"
+structopt = "0.2.16"
```
Comment (utopiabound) [inline iml-agent/Cargo.toml]: Is this supposed to be backwards?

Comment (jgrund) [reply iml-agent/Cargo.toml]: 0.2.17 has been yanked: https://crates.io/crates/structopt/0.2.17

File: iml-agent/Cargo.toml (hunk 1 above)
Comment (jgrund) [reply iml-agent/Cargo.toml]: I'll change to 0.9

Comment (jgrund) [inline iml-agent/Cargo.toml]:
```suggestion
reqwest = { version = "0.9", features = ["default-tls", "tls"] }
```

**Pull request merged**